daml/Marketplace/Custody/Model.daml

```diff
@@ -129,6 +128,25 @@ template LifecycleRequest
           archive assetDepositCid
           return (assetDepositCid', payoutInstructions)
 
+lookupOrInsertIndex : AssetDescription -> AssetDeposit -> Claims -> Update AssetDescription.Version
```

Comment:
This is just an idea, but consider whether it's worth coupling the `Index` and `AssetDescription` by referencing one in the other, via a `ContractId`. You can then ensure that they're not created separately and get out-of-sync.
I think you originally modified the `INdex.LookupOrInsert` to do the creation of both contracst in one step. I said we shouldn't do this because we didn't have an API to enforce it.
If you're bored enough, you can explore if there's a simple way to do this, without adding lots of boilerplate.
But this is just a suggestion - it's fine by me as-is here.

Reply:
So you mean just having a `ContractId AssetDescription` in the index? So every time an Index is created an AssetDescription contract ID is needed.

Reply:
@lucianojoublanc-da is it okay if AssetDescription has a reference to the Index and not the other way around? At lookup/insert, the index is looked up or created before the creation of the Asset Description (and that I think cannot be reversed).

Reply:
Give it a try. Don't see why not. The indices should be immutable.